iceoryx_hoofs/posix/vocabulary/source/file_path.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iox/file_path.hpp"
+#include "iox/filesystem.hpp"
+
+namespace iox
+{
+namespace details
+{
+bool file_path_does_contain_invalid_characters(const string<platform::IOX_MAX_PATH_LENGTH>& value) noexcept
+{
+    const auto valueSize = value.size();
+
+    for (uint64_t i{0}; i < valueSize; ++i)
+    {
+        // AXIVION Next Construct AutosarC++19_03-A3.9.1: Not used as an integer but as actual character
+        const char c{value[i]};
+
+        const bool isSmallLetter{'a' <= c && c <= 'z'};
```

Comment:
This cannot be constexpr since the function is not constexpr and inside a cpp file. Furthermore `c` is a const runtime variable.

Reply:
I think you misunderstood me. I meant something like
```
const bool isSmallLetter{ASCII_A <= c && c <= ASCII_Z};
```

Reply:
Alright, but I like `'a'` more since it is more straight forward. But I have no strong feelings about this.

I can change it if you give me thumbs up.